pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/JsonExtractIndexTransformFunction.java

```diff
@@ -90,15 +90,12 @@ public void init(List<TransformFunction> arguments, Map<String, ColumnContext> c
     }
     String resultsType = ((LiteralTransformFunction) thirdArgument).getStringLiteral().toUpperCase();
     boolean isSingleValue = !resultsType.endsWith("_ARRAY");
-    // TODO: will support array type; the underlying jsonIndexReader.getMatchingDocsMap supports the json path [*]
-    if (!isSingleValue) {
-      throw new IllegalArgumentException("jsonExtractIndex only supports single value type");
-    }
     if (isSingleValue && inputJsonPath.contains("[*]")) {
```

Comment:
Should we consider returning the first matching value when a SV matches multiple values?

Reply:
I personally would not prefer that, since `[*]` in the extract path typically inherently implies that the user expects an array of values.. but open to suggestions